### DISCUSSION TITLE
is it licensing safe to provide tempo datasource

### DISCUSSION THREAD

**Discussion by jiancaocn** [uptrace/uptrace#311] [General]
just for curiosity, tempo is AGPL v3, whether it's ok to provide a tempo datasource connection for uptrace Cloud
![image](https://github.com/uptrace/uptrace/assets/13940131/ca4acc9d-80a7-4c8e-add5-a37dbe9c0d84)

**Comment by vmihailenco** [comment]
I would assume so since it is just a Tempo-like API. We don't use any code from Grafana to achieve that.

PS Well, we actually do use Protobuf bindings from Tempo repository. It is not hard to replace it, but Uptrace is also AGPL-licensed so I guess we're fine for now.

**Reply by jiancaocn** [reply]
got this thread, https://opensource.stackexchange.com/questions/8201/if-i-use-a-public-api-endpoint-that-has-its-source-code-licensed-under-agpl-in-m
it seems ok if you just use the API
